api/src/org/apache/cloudstack/api/command/admin/vm/BulkDeployVMCmd.java

```diff
@@ -0,0 +1,268 @@
+            + "Should be passed only when VM is created from a zone with Basic Network support. Mutually exclusive with securitygroupids parameter")
+    private List<String> securityGroupNames;
+
+    @ACL
+    @Parameter(name = ApiConstants.AFFINITY_GROUP_IDS, type = CommandType.LIST, collectionType = CommandType.UUID, entityType = AffinityGroupResponse.class,
+            description = "comma separated list of affinity groups IDs that are going to be applied to the virtual machine. "
+            + "Mutually exclusive with affinitygroupnames parameter")
+    private List<Long> affinityGroupIds;
+
+    @ACL
+    @Parameter(name = ApiConstants.AFFINITY_GROUP_NAMES, type = CommandType.LIST, collectionType = CommandType.STRING, entityType = AffinityGroupResponse.class,
+            description = "comma separated list of affinity groups names that are going to be applied to the virtual machine. "
+            + "Mutually exclusive with affinitygroupids parameter")
+    private List<String> affinityGroupNames;
+
+    @Parameter(name = ApiConstants.ACCOUNT, type = CommandType.STRING, description = "an optional account for the virtual machine. Must be used with domainId.")
+    private String accountName;
+
+    @Parameter(name = ApiConstants.DOMAIN_ID, type = CommandType.UUID, entityType = DomainResponse.class, description = "an optional domainId for the virtual machine. If the account parameter is used, domainId must also be used.")
+    private Long domainId;
+
+    public Long getZoneId() {
+        return zoneId;
+    }
+
+    public Long getPodId() {
+        return podId;
+    }
+
+    public Long getClusterId() {
+        return clusterId;
+    }
+
+    public Long getServiceOfferingId() {
+        return serviceOfferingId;
+    }
+
+    public Long getTemplateId() {
+        return templateId;
+    }
+
+    public String getHypervisor() {
+        return hypervisor;
+    }
+
+    public List<Long> getNetworkIds() {
+        return networkIds;
+    }
+
+    public Long getDiskOfferingId() {
+        return diskOfferingId;
+    }
+
+    public List<Long> getSecurityGroupIds() {
+        return securityGroupIds;
+    }
+
+    public List<String> getSecurityGroupNames() {
+        return securityGroupNames;
+    }
+
+    public String getUserData() {
+        return userData;
+    }
+
+    public String getSshKeyPairName() {
+        return sshKeyPairName;
+    }
+
+    public List<Long> getAffinityGroupIds() {
+        return affinityGroupIds;
+    }
+
+    public List<String> getAffinityGroupNames() {
+        return affinityGroupNames;
+    }
+
+    public String getAccountName() {
+        return accountName;
+    }
+
+    public Long getDomainId() {
+        return domainId;
+    }
+
+    public Integer getCount() {
+        return count;
+    }
+
+    private void verifyInputs() {
+        if (count == null || count <= 0) {
```

Comment:
Can you consider using Preconditions check, or at least throw `ServerApiException(ApiErrorCode.PARAM_ERROR, ...` ?